code/functions/api/[[path]].js

```diff
@@ -0,0 +1,34 @@
+export async function onRequest(context) {
+  // Contents of context object
+  let {
+    request, // same as existing Worker API                           -> needed
+    env, // same as existing Worker API                               -> needed for different api backends than nuvla.io
+    params, // if filename includes [id] or [[path]]
+    waitUntil, // same as ctx.waitUntil in existing Worker API
+    next, // used for middleware or to fetch assets
+    data, // arbitrary space for passing data between middlewares
+  } = context;
+  const projectNameOnCFPages = 'nuvla-ui';
+  const url = new URL(request.url);
+
+  const firstPartOfHost = url.host.split('.')[0];
+
+  const procutionApiEndpoint = 'https://nuvla.io';
+
+  // try get api endpoint from environment or staging for preview depoyments, if present
+  let apiUrl =
+    firstPartOfHost === projectNameOnCFPages
+      ? procutionApiEndpoint
+      : env[firstPartOfHost] || env['staging'] || 'https://nuvla.io';
+
+  let response = await fetch(apiUrl + url.pathname, request);
+
+  // override base-uri for /api/cloud-entry-point responses
+  if (url.pathname.includes('cloud-entry-point')) {
+    let body = await response.json();
+    body = { ...body, 'base-uri': url.origin + '/api/' };
```

Comment:
base-uri is used to generate redirects server side e.g. (when you try to login with 2 factor authentication).
I wasn't able to login because of that. (Cross-Origin Request Blocked)

Reply:
Strange: I did this change because all requests to the backend after `/api/cloud-entry-point` went directly to `nuvla.io` and that was blocked because of cross-origin request. That's the same problem I had with Caddy when configuring too much...

But that shouldn't be a problem because I don't have to do that when I develop locally, then `base-uri` from `cloud-entry-point` is correctly set to `https://nui.localhost/api`.

I just tried it with 2FA: The request to `api/session` responds with location=﻿﻿﻿`﻿﻿https://nuvla.io/api/callback/b3e06829-112f-406c-9c45-1b21dbf76b4e/execute` on the preview URL, but the host should be from the preview URL.

On nui.localhost with caddy as reverse proxy it correctly sets the host to `nui.localhost`. I investigate later and perhaps I need a dev call session 😄